### ISSUE TITLE
create a pydpiper version of GNU parallel / sge_batch

### ISSUE THREAD

**Issue by mcvaneede** [Mouse-Imaging-Centre/pydpiper#97]

**Comment by gdevenyi** [comment]
Possible starting code https://github.com/jiahao/PySGE

**Comment by gdevenyi** [comment]
Followup here, @pipitone and I wrote a SGE/PBS version of sge_batch that is a bit more flexible than existing versions, we're actively developing.

I plan to add LSF support soon. If we can keep it reasonably general then perhaps using it would expand the reach of pydpiper

https://github.com/pipitone/qbatch

**Comment by bcdarwin** [comment]
Amusingly, in the last week or so we've been discussing how we should consider moving away from maintaining our own backend ... Pydpiper should really be about scripting pipelines.  I guess one thing qbatch doesn't seem to handle is individual jobs coming with their own time/RAM/disk/shm/tmp requirements; something implementing the proposed Common Workflow Language (or some other API on top of basic qsub, etc.) might be more suitable.

________________________________

**Comment by gdevenyi** [comment]
We're in progress of adding standard specifications of walltime and memory requirements, but then again, clusters like scinet don't honour standard PBS memory requirements.

It think no matter what method is implemented for a pipeline usually hackery is required to get it working on a specific site.

CWL looks interesting, and overkill if I want to get something working quickly

**Comment by pipitone** [comment]
Yeah, as I see it, qbatch is simply a more functional sge_batch, and not
a tool for running whole pipelines.  

I think for pipelines, it's likely best to move to nipype or frameworks
of that ilk. Speaking of which, have you guys considered joining forces
with nipype? 

Also, yeah CWL sounds a big pain in the butt for the kinds of iterative
pipeline work we're up against.

**Comment by gdevenyi** [comment]
@bcdarwin If you have specific needs that would make qbatch usable for pydpiper in the short-to-midterm, I'd be happy to move to quickly implement them :)

Otherwise, I hope you'd find it helpful for more generic work submission on SciNet or your local MiCe cluster.

**Comment by bcdarwin** [comment]
This would probably make more sense than continuing to maintain sge_batch in minc-stuffs as qbatch is more developed.

cc @mcvaneede

**Comment by mcvaneede** [comment]
Agreed! I'm all in favor for using shared programs :-)

**Comment by gdevenyi** [comment]
Awesome, @pipitone and I welcome issues/feature requests.

Development for my purposes has slowed down, it has all the features I need (until I add LSF support)

**Comment by gdevenyi** [comment]
Follow-up here, @pipitone and I would love to see any feature requests/comments on the existing implementation.

**Comment by gdevenyi** [comment]
Hi again,

We have a stable 1.0 release now of qbatch. https://github.com/pipitone/qbatch/releases

We're announcing it via various mailing lists and giving a SciNet users' group talk this week as well.
